src/main/java/com/bandwidth/sqs/publisher/MappingSqsMessagePublisher.java

```diff
@@ -0,0 +1,23 @@
+package com.bandwidth.sqs.publisher;
+
+import java.time.Duration;
+import java.util.Optional;
+
+import io.reactivex.Single;
+import io.reactivex.functions.Function;
+
+public class MappingSqsMessagePublisher<T, U> implements SqsMessagePublisher<T> {
+
+    private final Function<T, U> map;
+    private final SqsMessagePublisher<U> delegate;
+
+    public MappingSqsMessagePublisher(SqsMessagePublisher<U> delegate, Function<T, U> map) {
+        this.map = map;
+        this.delegate = delegate;
+    }
+
+    @Override
+    public Single<String> publishMessage(T body, Optional<Duration> maybeDelay) {
+        return Single.defer(() -> delegate.publishMessage(map.apply(body), maybeDelay));
```

Comment:
this defer makes it "cold", right? as in this won't happen until someone subscribes to the returned single? Should you do something to make sure it kicks off, or simply require the caller to subscribe? If the latter, probably provide some documentation